includes/class-sensei-course.php

```diff
@@ -100,8 +100,9 @@ public function __construct () {
         add_action('save_post', array( 'Sensei_Course', 'flush_rewrite_rules' ) );
 
 		// Allow course archive to be setup as the home page
-		add_action( 'pre_get_posts', array( $this, 'allow_course_archive_on_front_page' ) );
-
+		if ( (int) get_option( 'page_on_front' ) > 0 ) {
```

Comment:
Looking at the _current_ code in master, we have this test:

`$possible_home_page->ID != get_option( 'page_on_front' )`

So, essentially, if `get_option( 'page_on_front' )` doesn't contain something that can be a post ID then for our purposes here we aren't interested. Core does something else, but then it's not testing for the same thing. 

That said, I don't mind revising this to the simpler test you suggested if that's what you prefer :+1: